### DISCUSSION TITLE
Enhance docs - Setup without using snippets

### DISCUSSION THREAD

**Discussion by jtuchel** [chakra-ui/chakra-ui#9326] [Ideas]
### Description

Please enhance the docs to show how to setup Chakra without using snippets

### Problem Statement/Justification

I would like to install the library and keep everything up to date by updating the version

### Proposed Solution or API

Based on this page

https://www.chakra-ui.com/docs/get-started/frameworks/vite

there should be two sections

- With snippets
- Without snippets

**Comment by pgxor** [comment]
yes it confused me too that are snippets mandatory or optional?

**Comment by thesohailjafri** [comment]
So all components in **Layout and Typography** are available from the chakra UI package (core components)
and everything other than that from snippets cli

### Available components Out of the box
![image](https://github.com/user-attachments/assets/5f498ff7-8cf2-4553-946b-86290a08b98b)
![image](https://github.com/user-attachments/assets/72d941d8-e6ae-4b5f-b5ef-3be05b196985)

### Install components via snippets
![image](https://github.com/user-attachments/assets/0b06e86c-499c-4d26-bfa4-91814b52a7c7)

**Comment by veloware** [comment]
> So all components in **Layout and Typography** are available from the chakra UI package (core components) and everything other than that from snippets cli
> 
> ### Available components Out of the box
> ![image](https://private-user-images.githubusercontent.com/66732877/395168643-5f498ff7-8cf2-4553-946b-86290a08b98b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMTMwMTQsIm5iZiI6MTczNDAxMjcxNCwicGF0aCI6Ii82NjczMjg3Ny8zOTUxNjg2NDMtNWY0OThmZjctOGNmMi00NTUzLTk0NmItODYyOTBhMDhiOThiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE0MTE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4MWQwN2JlNzUyZGQ3ODY5ODVjNmEwNmZiMDc5ZDJiZTNhMzVhNzAwMDA3OGJkOTllNjM5OTg0OTdlNWM2ZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NifQZRDtlUUHuNDqlcVK57o121M_ytgvSP49dtrwkkU) ![image](https://private-user-images.githubusercontent.com/66732877/395168628-72d941d8-e6ae-4b5f-b5ef-3be05b196985.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMTMwMTQsIm5iZiI6MTczNDAxMjcxNCwicGF0aCI6Ii82NjczMjg3Ny8zOTUxNjg2MjgtNzJkOTQxZDgtZTZhZS00YjVmLWI1ZWYtM2JlMDViMTk2OTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE0MTE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4MGJkYzhmMTEyMTBmODJlNTk1ZDU0MDdiZTI0MTVhNGNjNDdkMWFmNjhkMGVkYTVhODM2YzExMWEzNjljNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vQMzbhx8lgetTsVJ66xhb96K-4UJK6_edl3nLsM329Y)
> 
> ### Install components via snippets
> ![image](https://private-user-images.githubusercontent.com/66732877/395168594-0b06e86c-499c-4d26-bfa4-91814b52a7c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMTMwMTQsIm5iZiI6MTczNDAxMjcxNCwicGF0aCI6Ii82NjczMjg3Ny8zOTUxNjg1OTQtMGIwNmU4NmMtNDk5Yy00ZDI2LWJmYTQtOTE4MTRiNTJhN2M3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE0MTE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlNTE0M2I4MTAyNTA2MTc5NTdkNTI0MGFkYjM0ZWM5MDUyODNiNmI5MWI1N2IyZTc3OWNmOGRkOTdlZDdlZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oYu5XD1gNSezMnS0mFl_rHWOK6Fh_vbkTp-GQ3gOUn8)

Thats not true, and that's kind of the issue regarding the confusion in the docs that has been raised multiple times, for example `Card` and `Input` are not snippets.

**Comment by segunadebayo** [comment]
I've started adding some docs around this. Open to contributions to further improve it.
https://github.com/chakra-ui/chakra-ui/commit/6ccf39f690a6d56b82b76d568f159aa1f78e8442

A good starting point is to add "Without Snippet" examples. Over time, we'll consider adding tabs to each example (Snippet / Without Snippet)

**Reply by jtuchel** [reply]
> I've started adding some docs around this. Open to contributions to further improve it. [6ccf39f](https://github.com/chakra-ui/chakra-ui/commit/6ccf39f690a6d56b82b76d568f159aa1f78e8442)
> 
> A good starting point is to add "Without Snippet" examples. Over time, we'll consider adding tabs to each example (Snippet / Without Snippet)

Thanks a lot! :)

So just to be clear, currently this is not possible, right? I installed `@chakra-ui/react @emotion/react` and simply tried this

```tsx
import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { Button, ChakraProvider } from '@chakra-ui/react';

createRoot(document.getElementById('root')!).render(
  <StrictMode>
    <ChakraProvider>
      <Button>Foo</Button>
    </ChakraProvider>
  </StrictMode>
);
```

 ( sandbox => https://stackblitz.com/edit/vitejs-vite-2vzw6yzp?file=src%2Fmain.tsx )

It seems Chakra needs the provider and the provider needs props. If the CLI is required at the moment are there any future plans?

**Comment by segunadebayo** [comment]
Snippets are optional copies of Provider, toaster, and a few composed pieces. Almost every component, including Card and Input, is exported from `@chakra-ui/react`.

Install the package, wrap with `ChakraProvider`, skip the CLI if you want.
